numpy/lib/tests/test_function_base.py

```diff
@@ -2966,7 +2966,13 @@ def test_basic(self):
         assert_equal(np.quantile(x, 0), 0.)
         assert_equal(np.quantile(x, 1), 3.5)
         assert_equal(np.quantile(x, 0.5), 1.75)
-
+    def test_corner(self):
+        a = True
+        q = False
+        assert_equal(np.quantile(a, q), 1.0)
+        a = np.array([False, True, True])
+        q = a
+        assert_array_equal(np.quantile(a, q), [0, 1, 1])
```

Comment:
Blank line after this please. `assert_array_equal` does not actually assert that the `arr.dtype` is equal, so you should assert that. Also please give the test a more descriptive name than "corner". The line `q = a` does not do anything, so just don't worry about it. The first two lines (a, and q) also might as well be directly in the `np.quantile` statement. Rather put `res = ...` and then assert `res.dtype`, etc.

Reply:
Updated the test, see if that looks better. Both Windows and Ubuntu with the code update are now consistent for quantile to the values that Ubuntu was before.